src/internal/lokiutil/testloki/testloki.go

```diff
@@ -32,13 +32,25 @@ type TestLoki struct {
 	lokiErrCh <-chan error
 }
 
+type Option func(config *map[string]any)
+
+func WithoutOldSampleRejection(config *(map[string]any)) {
+	(*config)["limits_config"].(map[string]any)["reject_old_samples"] = false
+}
+
+func WithCreationGracePeriod(d time.Duration) func(config *(map[string]any)) {
+	return func(config *(map[string]any)) {
+		(*config)["limits_config"].(map[string]any)["creation_grace_period"] = d
+	}
+}
+
```

Comment:
I would just set these to infinite in the default config.  That was what I had in mind when making the config file.